### PR TITLE
dist: build linux/darwin arm64 targets

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
Adding the linux/darwin arm64 targets, we start building arm64 for the server and will start adding the targets for the plugins for the ones we prepackage in the build

related to: https://github.com/mattermost/mattermost-server/pull/18491

#### Ticket Link
related to https://mattermost.atlassian.net/browse/DOPS-633

### CODE REVIEW

File: server/issue.go [hunk 1]
```diff
@@ -39,10 +39,11 @@ func makePost(userID, channelID, message string) *model.Post {
...
 
 func (p *Plugin) httpShareIssuePublicly(w http.ResponseWriter, r *http.Request) (int, error) {
-	requestData := model.PostActionIntegrationRequestFromJson(r.Body)
-	if requestData == nil {
+	var requestData model.PostActionIntegrationRequest
+	err := json.NewDecoder(r.Body).Decode(&requestData)
+	if err != nil {
 		return respondErr(w, http.StatusBadRequest,
-			errors.New("missing request data"))
+			errors.New("unmarshall the body"))
```
Comment (mickmister) [inline server/issue.go:46]: Should we wrap the error here?

File: server/subscribe.go [hunk 2]
```diff
@@ -587,7 +587,7 @@ func (p *Plugin) hasPermissionToManageSubscription(instanceID types.ID, userID,
 
 	switch cfg.RolesAllowedToEditJiraSubscriptions {
 	case "team_admin":
-		if !p.API.HasPermissionToChannel(userID, channelID, model.PERMISSION_MANAGE_TEAM) {
+		if !p.API.HasPermissionToChannel(userID, channelID, model.PermissionManageTeam) {
```
Comment (mickmister) [inline server/subscribe.go:590]: I wonder if this should be `HasPermissionToTeam` or if this still works like this

**Review by mickmister — APPROVED** [review]
LGTM

File: server/subscribe.go (hunk 2 above)
Comment (cpanato) [reply server/subscribe.go:590]: maybe @hanzei can comment this

File: server/issue.go (hunk 1 above)
Comment (cpanato) [reply server/issue.go]: done

File: server/subscribe.go (hunk 2 above)
Comment (hanzei) [reply server/subscribe.go:590]: I don't think the semantics for permissions have changed. Also, the server does fall back to `HasPermissionToTeam`: https://github.com/mattermost/mattermost-server/blob/21639f9c8730b2f49ee52322acd3acdbf64ac6d9/app/authorization.go#L190

**Review by hanzei — APPROVED** [review]

**Pull request merged**